src/parser.rs

```diff
@@ -94,10 +95,12 @@ fn spans(style: Style) -> impl Fn(&[u8]) -> IResult<&[u8], (Spans<'static>, Styl
         let (s, _) = opt(tag("\n"))(s)?;
         let mut spans = Vec::new();
         let mut last = style;
-        loop {
-            let (s, span) = span(last)(text)?;
-            last = span.style;
-            spans.push(span);
+        while let Ok((s, span)) = span(last)(text) {
+            // don't include empty spans, provided we have at least one
```

Comment:
I think we should keep the empty spans. Since in an ascii art there may be empty lines which need to be displayed properly.